modules/core/src/main/java/org/apache/ignite/internal/replicator/ZonePartitionId.java

```diff
@@ -93,6 +95,19 @@ public static ZonePartitionId fromString(String str) {
         return new ZonePartitionId(Integer.parseInt(parts[0]), Integer.parseInt(parts[1]));
     }
 
+    /**
+     * Creates a new object if this one has a defined table id or returns itself if it does not have a value of the table id.
+     *
+     * @return Pure zone partition id.
+     */
+    public ZonePartitionId purify() {
+        if (tableId == 0) {
```

Comment:
I would change this method and like this
```
    /**
     * Creates a new {@link ZonePartitionId} without {@code tableId} if {@code zonePartitionId} has a specified {@code tableId} 
     * or returns provided {@code zonePartitionId} otherwise.
     *
     * @param zonePartitionId Zone partition id.
     * @return Zone partition id without specified {@code tableId}.
     */
    public static ZonePartitionId resetTableId(ZonePartitionId zonePartitionId) {
        if (zonePartitionId.tableId() == 0) {
            return zonePartitionId;
        }

        return new ZonePartitionId(zonePartitionId.zoneId(), zonePartitionId.tableId());
    }
```

Reply:
Ok